utils/sccli.py

```diff
@@ -0,0 +1,224 @@
+               " kube-controller-manager kube-scheduler kube-proxy kubelet")
+        if not system("systemctl is-active etcd")[2]:
+            system("systemctl stop etcd")
+    elif service_name == "openshift":
+        system("systemctl disable openshift")
+        if not system("systemctl is-active openshift")[2]:
+            system("systemctl stop openshift")
+        system("systemctl reset-failed openshift")
+    elif service_name == "docker":
+        if not system("systemctl is-active docker")[2]:
+            system("systemctl stop docker")
+        system("systemctl reset-failed docker")
+
+def setup_kube_service_account_key():
+    system("mkdir -p /etc/pki/kube-apiserver/")
+    system("openssl genrsa -out /etc/pki/kube-apiserver/serviceaccount.key 2048")
+    system("sed -i.back '/KUBE_API_ARGS=*/c\KUBE_API_ARGS="
+           "\"--service_account_key_file=/etc/pki/kube-apiserver/serviceaccount.key\"'"
+           " /etc/kubernetes/apiserver")
+    system("sed -i.back '/KUBE_CONTROLLER_MANAGER_ARGS=*/c\KUBE_CONTROLLER_MANAGER_ARGS="
+           "\"--service_account_private_key_file=/etc/pki/kube-apiserver/serviceaccount.key\"'"
+           "/etc/kubernetes/controller-manager")
+
+# This is not used "need discussion"
+def vagrant_box_variant():
+    try:
+        with open("/etc/os-release") as fh:
+            for line in fh.readlines():
+                if "Container Development Kit" in line:
+                    return False
+        return True
+    except IOError as err:
+        return (err, 37)
+
+def pull_openshift_images():
+    docker_registry = os.getenv('DOCKER_REGISTRY', DOCKER_REGISTRY)
+    image_name = os.getenv('IMAGE_NAME', IMAGE_NAME)
+    image_tag = os.getenv('IMAGE_TAG', IMAGE_TAG)
+    if system(('sed -i.back "/^IMAGE=*/cIMAGE=\"%s/%s\:%s\""'
+            ' /etc/sysconfig/openshift_option') % (docker_registry, image_name, image_tag))[2]:
+        return ("Permisison denined: /etc/sysconfig/openshift_option", 37)
+    if system("docker pull %s/%s:%s" % (docker_registry, image_name, image_tag))[2]:
+        return ("%s/%s:%s Not Pulled" % (docker_registry, image_name, image_tag), 111)
+    if system("docker pull %s/%s-haproxy-router:%s" % (docker_registry, image_name, image_tag))[2]:
+        return ("%s/%s-haproxy-router:%s Not Pulled" % (docker_registry, image_name, image_tag), 111)
+    if system("docker pull %s/%s-deployer:%s" % (docker_registry, image_name, image_tag))[2]:
+        return ("%s/%s-deployer:%s Not Pulled" % (docker_registry, image_name, image_tag), 111)
+    if system("docker pull %s/%s-docker-registry:%s" % (docker_registry, image_name, image_tag))[2]:
+        return ("%s/%s-docker-registry:%s Not Pulled" % (docker_registry, image_name, image_tag), 111)
+    if system("docker pull %s/%s-sti-builder:%s" % (docker_registry, image_name, image_tag))[2]:
+        return ("%s/%s-sti-builder:%s Not Pulled" % (docker_registry, image_name, image_tag), 111)
+    return ('', 0)
+
```

Comment:
> Looks like we are not checking if the images already present

No Since first draft is equivant to what sccli provide and till now sccli.sh doesn't provide a way to check if images already present.